extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs

```diff
@@ -25,15 +38,85 @@ public static IHostBuilder ConfigureAspNetCoreIntegration(this IHostBuilder buil
             builder.ConfigureWebHostDefaults(webBuilder =>
             {
                 webBuilder.UseUrls(HttpUriProvider.HttpUriString);
+
                 webBuilder.Configure(b =>
                 {
+                    b.UseRouting();
                     b.UseMiddleware<WorkerRequestServicesMiddleware>();
+
                     // TODO: provide a way for customers to configure their middleware pipeline here
+
+                    b.UseEndpoints(endpoints =>
+                    {
+                        string scriptRoot = Environment.GetEnvironmentVariable(FunctionsApplicationDirectoryKey) ??
```

Comment:
Based on previous conversations, I think @brettsam was planning on doing that.

Reply:
Moved things to use a DataSource now. It does contain it a lot nicer.

I still have everything synchronous... the asynchronous nature of loading the routes made me nervous that we could run into a race where a request could sneak in before the routes were fully built and you'd get a null endpoint on the HttpContext.

Thoughts on that?

Reply:
Yeah, that race condition is possible, especially since cold start is much more common with functions than a traditional AspNetCore app.

As for async with sync, it doesn't bother me much there. It will be called once, so not much of an impact.